zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java

```diff
@@ -890,16 +890,15 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
 
       String noteId = context.getJobDetail().getJobDataMap().getString("noteId");
       Note note = notebook.getNote(noteId);
-      note.runAll();
 
-      while (!note.isTerminated()) {
-        try {
-          Thread.sleep(1000);
-        } catch (InterruptedException e) {
-          logger.error(e.toString(), e);
-        }
+      if (note.isRunningOrPending()) {
+        logger.info("execution of the cron job is skipped because there is a running or pending " +
+            "paragraph (note id: {})", noteId);
+        return;
```

Comment:
so if someone is also running the notebook manually when the scheduler kicks off, would that block the scheduler?

Reply:
if so, this is a behavior change that is worthwhile to document

Reply:
> so if someone is also running the notebook manually when the scheduler kicks off, would that block the scheduler?

Yes. I think it's OK because there's no need that notebooks which are executed manually are kicked by the cron scheduler.

> if so, this is a behavior change that is worthwhile to document

I'll add the explanation to document.

Thanks.